lib/linked_list.rb

```diff
@@ -7,39 +7,70 @@ def initialize
     @head = nil
   end
 
-  # Time complexity - ?
-  # Space complexity - ?
+  # Time complexity - O(1)
+  # Space complexity - O(1)
   def add_first(data)
-
+    @head = Node.new(data, head)
   end
 
-  # Time complexity - ?
-  # Space complexity - ?
+  # Time complexity - O(1)
+  # Space complexity - O(1)
   def get_first
+    return @head if @head.nil?
```

Comment:
more readable to have

```suggestion
    return nil if @head.nil?
```